File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-java/src/br/edu/ifpb/padroes/diegofrazao/fabricas/FabricaFiat.java
package br.edu.ifpb.padroes.diegofrazao.fabricas;

import br.edu.ifpb.padroes.diegofrazao.carros.Carro;
import br.edu.ifpb.padroes.diegofrazao.carros.Palio;

public class FabricaFiat implements FabricaDeCarro {
    @Override
    public Carro criarCarro() {
        return new Palio();
    }
}

File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-typescript/fabricas/FabricaFiat.ts
import { Carro } from '../carros/Carro'
import { Palio } from '../carros/Palio'
import { FabricaDeCarro } from './FabricaDeCarro'

export class FabricaFiat extends FabricaDeCarro {

	public criarCarro(): Carro {
		return new Palio();
	}

}